src/core/linter-rules/no-unused-vars.js

```diff
@@ -0,0 +1,66 @@
+// @ts-check
+/**
+ * Linter rule "no-unused-vars".
+ *
+ * Checks that an variable is used if declared (the first use is treated as
+ * declaration).
+ */
+import LinterRule from "../LinterRule.js";
+import { lang as defaultLang } from "../l10n.js";
+import { norm } from "../utils.js";
+
+const name = "no-unused-vars";
+
+const meta = {
+  en: {
+    description: "Variable was defined, but never used.",
+    howToFix:
+      "Add a `data-ignore-unused` attribute to the `<var>`, or consider using `<em>`/ `<code>`.",
+    help: "See developer console.",
+  },
+};
+// Fall back to english, if language is missing
+const lang = defaultLang in meta ? defaultLang : "en";
+
+/**
+ * @param {*} _
+ * @param {Document} doc
+ * @return {import("../LinterRule").LinterResult}
+ */
+function linterFunction(_, doc) {
+  const offendingElements = [];
+
+  for (const section of doc.querySelectorAll("section")) {
```

Comment:
Should we pick only those sections that contain algorithms?
See: https://github.com/w3c/payment-request/blob/e9515622/index.html#L280

Reply:
I think so... we might need to test out a few specs.